File: Assignment/Abstraction/Statements/ElseIfStatementNode.cs
using System.Collections.Generic;

namespace Assignment.Abstraction.Statements
{
    internal struct ElseIfStatementNode : IBooleanStatementNode
    {
        public ElseIfStatementNode(IEnumerable<INode> body, INode expression)
        {
            this.Body = body;
            this.Expression = expression;
        }

        public IEnumerable<INode> Body { get; }
        public INode Expression { get; }

        public string DisplayName()
        {
            return nameof(ElseIfStatementNode);
        }
    }
}

File: src/main/java/ce305/abstraction/statements/ElseIfStatementNode.java
package ce305.abstraction.statements;

import java.util.Collection;
import java.util.List;
import java.util.UUID;

import ce305.abstraction.INode;

public class ElseIfStatementNode extends INode {
    public final List<INode> body;
    public final INode child;
    public final INode expression;

    public ElseIfStatementNode(List<INode> body, INode expression, INode child) {
        super();
        this.body = body;
        this.child = child;
        this.expression = expression;
    }

    public ElseIfStatementNode(List<INode> body, INode expression, INode child, UUID id) {
        super(id);
        this.body = body;
        this.child = child;
        this.expression = expression;
    }

    @Override
    public String getDisplayName() {
        return ElseIfStatementNode.class.getSimpleName();
    }
}